modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/MuleArtifactResourcesRegistry.java

```diff
@@ -194,6 +196,20 @@ public Builder withAdditionalResourceDirectory(String resourceDirectory) {
       return this;
     }
 
+    /**
+     * An action to perform on the classloader of the artifact that is created.
+     *
+     * @since 4.7.0
+     *
+     * @param actionOnMuleArtifactClassloader the action to be performed.
+     *
+     * @return the current builder.
+     */
+    public Builder withActionOnMuleArtifactClassloader(Consumer<ClassLoader> actionOnMuleArtifactClassloader) {
```

Comment:
missing `Creation` at the end, or something that identifies the action that triggers the callback